rudder-core/src/main/scala/com/normation/rudder/batch/AsyncDeploymentAgent.scala

```diff
@@ -162,14 +157,50 @@ final class AsyncDeploymentAgent(
     )
   }
 
-  override protected def lowPriority = {
+  // That flag file is used to determine if a policy update was running when the webapp stopped so we can launch a new one when starting
+  val policyUpdateRunningFlagPath = "/opt/rudder/etc/policy-update-running"
+
+  private[this] def createFlagFile = {
+    val file =  new File(policyUpdateRunningFlagPath)
+    try {
+        file.createNewFile()
+        logger.debug(s"Flag file '${policyUpdateRunningFlagPath}' created")
+    } catch {
+      // Exception while checking the file existence
+      case e : Exception =>
+        logger.error(s"An error occurred while creating flag file '${policyUpdateRunningFlagPath}', cause is: ${e.getMessage}")
+    }
+  }
 
+  private[this] def deleteFlagFile = {
+    val file =  new File(policyUpdateRunningFlagPath)
+    try {
+      if (file.delete) {
+        // Deleted, come back to normal
+        logger.debug(s"Flag file '${policyUpdateRunningFlagPath}' successfully removed")
+      } else {
+        // File could not be deleted, seek for reason
+        if(!file.exists()) {
+          logger.warn(s"Flag file '${policyUpdateRunningFlagPath}' could not be removed as it does not exists anymore")
+        } else {
+          logger.error(s"Flag file '${policyUpdateRunningFlagPath}' could not be removed, you may have to remove it manually, cause is: Permission denied or someone is actually editing the file")
+         }
+      }
+    } catch {
+      // Exception while checking the file existence
+      case e : Exception =>
+        logger.error(s"An error occurred while deleting flag file '${policyUpdateRunningFlagPath}', cause is: ${e.getMessage}")
+    }
+  }
+  override protected def lowPriority = {
 
     //
     // Start a new deployment
     //
     case AutomaticStartDeployment(modId, actor) => {
       implicit val a = actor
+      createFlagFile
```

Comment:
What's bothering me with stopping the process is that it prevents generation...

And so it would prevent new policies to be generated in a normal case (if the flag creation failed)